mods/flowers/init.lua

```diff
@@ -204,38 +204,34 @@ minetest.register_node("flowers:mushroom_brown", {
 
 -- Mushroom spread and death
 
+function flowers.mushroom_spread(pos, node)
+	if minetest.get_node_light(pos, nil) == 15 then
+		minetest.remove_node(pos)
+		return
+	end
+	local positions = minetest.find_nodes_in_area_under_air(
+		{x = pos.x - 1, y = pos.y - 2, z = pos.z - 1},
+		{x = pos.x + 1, y = pos.y + 1, z = pos.z + 1},
+		{"group:soil", "group:tree"})
+	if #positions == 0 then
+		return
+	end
+	local pos2 = positions[math.random(#positions)]
+	pos2.y = pos2.y + 1
+	if minetest.get_node_light(pos, 0.5) < 4 and
```

Comment:
<=3 is clearer (and no slower) because it makes clear that 3 is the maximum light level mushrooms can tolerate, this is useful information to know and more useful than knowing what level kills nushrooms.